### PR TITLE
Sr v1 workspace integration tests

### PR DESCRIPTION
**Pull request by s-rubenstein**
<!--
Hello, friend!
Remember to mention what sort of testing/verification you performed in the course of building this PR, i.e. checking functionality in the browser locally.
Thanks!
--!>

### CODE REVIEW

File: integration-tests/utils/integration-helpers.js
Comment (jdcanas) [inline integration-tests/utils/integration-helpers.js:79]: nit: It might be worth making a constants file (or use an existing one) for these access levels, see https://github.com/DataBiosphere/terra-ui/pull/2500/files#diff-ada5c6a1752c4db6c566600346028c476c874042c68d93f0d187b2e0a3a07e04R110 as well. I dislike scattered magic strings with external dependencies. We are trying to do the same for Leo strings in terra-ui too...

File: integration-tests/tests/run-notebook.js
Comment (jdcanas) [inline integration-tests/tests/run-notebook.js:93]: TOL: is it worth calling `testRunNotebook` `testRunNotebookWithV2Workspace` ?

**Review by jdcanas — APPROVED** [review]
Looks good overall when they pass, few nits. Worth mentioning I haven't had a chance to do much beyond mess with the tests a bit locally, so I'd take my approval with a grain of salt. 

As an aside, we are currently doing a large revamp of notebooks, so I expect all notebook related tests to change in terms of the UI flow. I don't see this logic changing at all though, it seems modular enough to be adapted easily.

File: integration-tests/tests/run-notebook.js
Comment (s-rubenstein) [reply integration-tests/tests/run-notebook.js:93]: I am going to wait on that until CI takes a look! My thought is no, because it is whatever the default for creating a workspace is, not explicitly v2.

File: integration-tests/utils/integration-helpers.js
Comment (s-rubenstein) [reply integration-tests/utils/integration-helpers.js:79]: Agreed. I can add that.

File: integration-tests/tests/run-notebook.js
Comment (s-rubenstein) [inline integration-tests/tests/run-notebook.js:16]: This was bumped up due to an issue with timing on asynchronous permissions updates (namely google granting bucket access)

File: integration-tests/utils/integration-helpers.js
Comment (s-rubenstein) [reply integration-tests/utils/integration-helpers.js:79]: I might add that as a separate tech debt ticket to avoid scope creep. We have a few of these scattered throughout the code-base as well, and I would want to hit those at the same time: https://broadworkbench.atlassian.net/browse/CA-1360

File: integration-tests/tests/run-workflow.js
Comment (andy7i) [inline integration-tests/tests/run-workflow.js:57]: (just curious) could we add a `if (!newEntity)` here? to skip some of this cleanup?

File: integration-tests/utils/integration-helpers.js
Comment (andy7i) [inline integration-tests/utils/integration-helpers.js]: is it worth logging the user's email here?

**Review by andy7i — APPROVED** [review]
👍 , couple questions but not blocking. Did not try running tests locally

File: integration-tests/tests/run-workflow.js
Comment (s-rubenstein) [reply integration-tests/tests/run-workflow.js:57]: Yes, I can add that. It might speed up the tests in some places.

File: integration-tests/utils/integration-helpers.js
Comment (s-rubenstein) [reply integration-tests/utils/integration-helpers.js]: Sure! Why not!

**Comment by zarsky-broad** [comment]
Can you give me a heads up when this is almost ready to merge so I can do a review pass? <3

**Comment by s-rubenstein** [comment]
@zarsky-broad it is almost ready to merge. Just waiting for another review from @gpolumbo-broad

File: integration-tests/tests/run-workflow.js
Comment (zarsky-broad) [inline integration-tests/tests/run-workflow.js]: Remove these?

File: integration-tests/tests/run-notebook.js
Comment (zarsky-broad) [inline integration-tests/tests/run-notebook.js]: I'd like to see this defined above its usages. Also, I'm not sure that `newEntity` is the right name for this, maybe like `isWorkspaceSingleUse`?

File: integration-tests/tests/run-workflow.js
Comment (zarsky-broad) [inline integration-tests/tests/run-workflow.js]: As in the other file, I'd prefer to see this defined above where it's invoked.

**Review by zarsky-broad — COMMENTED** [review]
I don't know what this v1 workspace thing is about, so my comments aren't about that, just a few things. I think it might be worth making sure teams know that this ~doubles the amount of time it takes tests to run, I might think there was something wrong otherwise.

File: integration-tests/tests/run-notebook.js
Comment (s-rubenstein) [reply integration-tests/tests/run-notebook.js]: Done

File: integration-tests/tests/run-workflow.js
Comment (s-rubenstein) [reply integration-tests/tests/run-workflow.js]: Done

File: integration-tests/tests/run-workflow.js
Comment (s-rubenstein) [reply integration-tests/tests/run-workflow.js]: Done

**Comment by s-rubenstein** [comment]
@zarsky-broad I wouldn't expect time to double? Since don't we run tests in parallel? It might be a bit longer, but I would expect it to be small, not huge. (At least, `integration-tests/yarn test` runs them in parallel)

**Comment by zarsky-broad** [comment]
> @zarsky-broad I wouldn't expect time to double? Since don't we run tests in parallel? It might be a bit longer, but I would expect it to be small, not huge. (At least, `integration-tests/yarn test` runs them in parallel)

The issue is that on circle we only have 2 cpu threads to work with, so our parallelism is very low; I'm working on another solution for that now. Comparing the time taken for the tests on this branch vs others, I'm seeing ~10 mins vs ~6:30

File: integration-tests/tests/run-notebook.js
Comment (gpolumbo-broad) [reply integration-tests/tests/run-notebook.js:93]: I suggest leaving a comment to explain this.  I think the point here is that we expect `testRunNotebook` to run in a Workspace created via the latest mechanism (might be v1, v2 or some future version) and `testRunNotebookWithV1Workspace` is here while we transition to PPW, and after we've completed the PPW migration, we should no long need it.  I might also suggest creating a Jira issue in the PPW Cleanup epic to come back and clean-up/delete the tests no longer being used.

File: integration-tests/tests/run-workflow.js
Comment (gpolumbo-broad) [inline integration-tests/tests/run-workflow.js:100]: Same in this file, if one of these tests is intended to be around forever and one is only intended to be around until some external condition is met (i.e., PPW migration is done) then let's comment as such and leave a marker for how to find this and how to identify when it can be safely removed.

Comment (s-rubenstein) [reply integration-tests/tests/run-workflow.js:100]: Filed https://broadworkbench.atlassian.net/browse/CA-1366

**Review by gpolumbo-broad — APPROVED** [review]

**Pull request merged**
